Lambdas/GHHooks/Models/+SemanticVersion.swift

```diff
@@ -0,0 +1,111 @@
+import SwiftSemver
+
+enum SemVerBump {
+    case releaseStage
+    case major
+    case minor
+    case patch
+}
+
+extension SemanticVersion {
+    func next(_ bump: SemVerBump) -> SemanticVersion? {
+        var version = self
+
+        if version.prereleaseIdentifiers.isEmpty {
+            switch bump {
+            case .releaseStage:
+                return nil
+            case .patch:
+                version.patch += 1
+            case .minor:
+                version.patch = 0
+                version.minor += 1
+            case .major:
+                version.patch = 0
+                version.minor = 0
+                version.major += 1
+            }
+        } else {
+            guard version.prereleaseIdentifiers.count < 4 else {
+                /// Shouldn't have more than 3 identifiers, otherwise unexpected by this code.
+                return nil
+            }
+            /// At this point we are guaranteed 1/2/3 `prereleaseIdentifiers`.
+
+            guard !version.prereleaseIdentifiers[0].allSatisfy(\.isNumber) else {
+                /// Identifiers should be like `["alpha", "1"]`.
```

Comment:
Might be worth defining an `enum ReleaseStage { case alpha, beta, rc }` type and checking for `ReleaseStage(rawValue: ...)` instead, come to think of it.

Reply:
It was like that but i didn't see a value in keeping the enum 🤔

Reply:
I guess i'll do it if you say so ... i don't really have a preference

Reply:
I do prefer it, though I'm too annoyed at myself to try to explain why just now 🤣